src/views/TodoView.vue

```diff
@@ -0,0 +1,329 @@
+<script setup lang="ts">
+import TableItem from "../components/TableItem.vue";
+import { computed, onMounted, ref } from "vue";
+import { ResponseData, TaskInfo, debounce } from "../util/shared";
+import {
+  deletePost,
+  getPosts,
+  postPost,
+  updatePost,
+} from "../util/HttpRequests";
+import FlexCard from "../components/FlexCard.vue";
+import FlexList from "../components/FlexList.vue";
+import FlexText from "../components/FlexText.vue";
+import AddTodo from "../components/AddTodo.vue"
+import { useRouter } from "vue-router";
+import IconButton from "../components/IconButton.vue";
+
+const DEBOUNCE_TIME = 250;
+
+let oldGet: ResponseData[]; // Keep a backup to restore from
+const router = useRouter()
+
+const sortingOrder = ref<SORT_OPTIONS>("task_dsc");
+
+const todoData = ref<ResponseData[]>([]);
+
+const filterInput = ref("");
+const addText = ref("");
+
+type SORT_OPTIONS = "task_asc" | "task_dsc" | "complete" | "incomplete";
+
+const sortingOptions: { text: string; value: SORT_OPTIONS }[] = [
+  { text: "Tasks (dsc)", value: "task_dsc" },
+  { text: "Tasks (asc)", value: "task_asc" },
+  { text: "Complete", value: "complete" },
+  { text: "Incomplete", value: "incomplete" },
+];
+
+const todoFiltered = computed(() => {
+  return [...todoData.value].filter((value) => {
+    // Copy or we end up blowing the stack
+    return value.todoName
+      .toLowerCase()
+      .includes(filterInput.value.toLowerCase());
+  });
+});
```

Comment:
Can be simplified by removing braces and return statements, and I'm not sure why you're using spread here

Reply:
I used copied the values because for some reason there was an odd interaction with the ref causing endless recursive calls.

Reply:
I really am not sure why that would be happening I'd love if you could show me that

Reply:
Weird, it's no longer an issue. I was probably updating the ref this uses somewhere causing endless calls...